Automattic/wp-calypso
# Remove lib/form-state/store

Removes the `lib/form-state/store` library and its only usage in passwordless signup.

The history of the `store` library is that in 2015 (pre-OSS repo) someone started to extract parts of `lib/form-state` to a separate library and never finished the project.

The usage in passwordless signup can be replaced with trivial local state for the `code` and `email` fields. I'm making these fields fully controlled (`value` prop) and removing the unneeded hidden `code` field from the form that contains only email address.

**How to test:**
1. Go to `/start/simple` as logged-out (you'll be redirected to `/` if logged in)
2. Create a passwordless user by entering email and the verification code

Not that the passwordless flow is already a bit broken in production: it creates the user successfully, but doesn't create a site and doesn't 100% setup login cookies, so the final redirect to `/` ends up at a logged out page. But the form flow itself should work.
